tests/testthat/tests-clone_instat_calculation.R

```diff
@@ -0,0 +1,76 @@
+library(testthat)
+library(R6)
+library(instatCalculations)
+
+# Cloner class that includes the clone_instat_calculation method
+Cloner <- R6Class("Cloner",
+                  public = list(
+                    clone_instat_calculation = function(curr_instat_calculation, ...) {
+                      new_manips <- lapply(curr_instat_calculation$manipulations, function(x) self$clone_instat_calculation(x))
+                      new_subs <- lapply(curr_instat_calculation$sub_calculations, function(x) self$clone_instat_calculation(x))
+                      instat_calculation$new(
+                        function_exp = curr_instat_calculation$function_exp,
+                        type = curr_instat_calculation$type,
+                        name = curr_instat_calculation$name,
+                        result_name = curr_instat_calculation$result_name,
+                        manipulations = new_manips,
+                        sub_calculations = new_subs,
+                        calculated_from = curr_instat_calculation$calculated_from,
+                        save = curr_instat_calculation$save
+                      )
+                    }
+                  )
+)
+
+test_that("clone_instat_calculation clones a basic instat_calculation", {
+  original_calc <- instat_calculation$new(
+    function_exp = quote(a + b),
+    type = "standard",
+    name = "calc1",
+    result_name = "result1",
+    calculated_from = list("a", "b"),
+    save = TRUE
+  )
+  
+  cloner <- Cloner$new()
+  cloned_calc <- cloner$clone_instat_calculation(original_calc)
```

Comment:
The issue is that you're testing `Cloner`, so there is mocking occurring here I think.
We want to have

`cloner <- DataBook$new()`

because we want to be testing the DataBook R6 class, not Cloner.